### ISSUE TITLE
Shorter and less technical labels for device picker

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/spreed#13987]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Feature request

**Which Nextcloud Version are you currently using:**
Nextcloud Hub 9 (31.0.0 dev)

**Is your feature request related to a problem? Please describe.**
This is an issue especially with microphone as it’s less obvious to check than the camera.

My list looks like this:
- Tiger Lake-LP Smart Sound Technology Audio Controller Digital Microphone
- Monitor of Tiger Lake-LP Smart Sound Technology Audio Controller Headphones
- Tiger Lake-LP Smart Sound Technology Audio Controller Headphones Stereo Microphone
- None

We already cut it a bit via `name-parts__last` but that always only seems to take the last word (at least in my case), so there’s 2 entries with "Microphone".

![Image](https://github.com/user-attachments/assets/2c9e52ae-0b6b-441a-8dbe-0abef4c2503c)

**Describe the solution you'd like**
What would be ideal is:
- Internal microphone
- Headphones microphone
- None

The "Monitor of" device basically never makes sense, I don’t know what it is. And the whole part of `Tiger Lake-LP Smart Sound Technology Audio Controller` is just technical jargon which does not need to be shown.

**Describe alternatives you've considered**
A very simple improvement would be to organize what is shown in front and behind, e.g. in my case it would already be a massive improvement to change it around like this:
- Digital Microphone (Tiger Lake-LP Smart Sound Technology Audio Controller)
- Headphones (Monitor of Tiger Lake-LP Smart Sound Technology Audio Controller) – although this one seems useless?
- Headphones Stereo Microphone (Tiger Lake-LP Smart Sound Technology Audio Controller)
- None

**Comment by miaulalala** [comment]
Ooof that's gonna be a hard one as there's so many different device types and so many different options of what people want to use. Not even Chrome hides those labels and simply shows the device name.

**Comment by jancborchardt** [comment]
@nickvergessen yep – indeed the OS display is still the nicest, whereas the other ones are basically the same (except Chrome). Is it possible to get the OS display names somehow?

**Comment by nickvergessen** [comment]
Not really, we just get "a Label":
https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/enumerateDevices

And if you look closely. the OS also has those labels, it's just prefixing it with some more details.
I assume somewhere in Gnome, Kernel or whatever there is a huuuuuuge list of devices that groups each into a type and then the type can be used to prefix the label and add an icon?

**Comment by jancborchardt** [comment]
@ShGKme thanks for the info! :) 2 issues with that though:
- This is not necessarily possible (or easily possible) on every system. For example on GNOME/Fedora I can not find an easy way to do it
- Even if this can be done, every single person has to fix it for themselves, and only a tiny percentage of people would ever adjust this.

The screenshot that @nickvergessen posted of the OS seems a nice solution, just prepending further info:

![Image](https://github.com/user-attachments/assets/f3130f89-162e-4261-8a1f-69545f2006e5)
